.github/workflows/codecov.yaml

```
@@ -36,7 +36,7 @@ jobs:
       - name: Set up Go 1.x
         uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
         with:
-          go-version: 1.17
+          go-version: 1.18
```

> > I wanted only to fix the typescript part as it was mentioned as a blocker. But yeap, I think I can dig a little bit an check the coverage too.
> 
> @rm3l I've bumped up the `go-version` used inside the `.github/workflows/codecov.yaml` to use `1.18`. This way we are aligning it with the `setup-go` action inside the [ci.yaml](https://github.com/devfile/api/blob/main/.github/workflows/ci.yaml?rgh-link-date=2023-11-10T11%3A54%3A05Z#L38-L42) where we [run the codecov too](https://github.com/devfile/api/blob/main/.github/workflows/ci.yaml?rgh-link-date=2023-11-10T11%3A54%3A05Z#L66-L67).
> 
> I'd say that bumping up to `1.19` is more related to the scope of #1237. wdyt?

Yup, I agree.

---

I think it also makes sense to use the same version declared in `go.mod` and avoid repeating the go version here. So I'd suggest using the [`go-version-file`](https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file) property instead:

```suggestion
          go-version-file: go.mod
```